### DISCUSSION TITLE
the batch size argument of global_max_pool

### DISCUSSION THREAD

**Discussion by 595693085** [pyg-team/pytorch_geometric#7256] [General]
Hi,
when the node feature and edge index are generated dynamically in the model "forward" function (for example, they are generated by the first half of the network), it is unable to package them in a Data class, so if i want use "global_max_pool", there no "batch" could be parsed from the Data class, but the function need to add the argument, so how to use "global_max_pool" in this case. Thank you.

**Comment by rusty1s** [comment]
You would in this case be responsible to create/update the `batch` argument as well. If you share some details, I can help you further.
